bindings/rust/s2n-tls-sys/build.rs

```diff
@@ -250,6 +245,49 @@ fn build_cmake() {
     }
 }
 
+#[derive(PartialEq, Eq, PartialOrd, Ord)]
+struct Libcrypto {
+    version: String,
+    link: String,
+    include: String,
+    root: String,
+}
+
+impl Default for Libcrypto {
+    fn default() -> Self {
+        let mut results = vec![];
+
+        for (name, value) in std::env::vars() {
+            if let Some(version) = name.strip_prefix("DEP_AWS_LC_") {
+                if let Some(version) = version.strip_suffix("_INCLUDE") {
+                    let version = version.to_string();
```

Comment:
I'm a bit confused why this loop is needed in order to find the highest version. Doesn't DEP_AWS_LC_INCLUDE come from the aws-lc-sys crate? Why would it define environment variables for multiple versions of AWS-LC?

Reply:
So aws-lc-sys prefixes all of their artifacts with the specific version of the crate. In https://github.com/aws/aws-lc-rs/pull/277 we added the `links = "aws_lc_X_X_X"` property to the Cargo.toml which tells cargo that it links against a specific C library.

Since we don't know exactly what version will be resolved by semver, we have to iterate and pick the correct one.

Reply:
https://doc.rust-lang.org/cargo/reference/build-scripts.html#the-links-manifest-key

Reply:
Got it ok, thanks!